lib/publiccloud/basetest.pm

```diff
@@ -107,39 +107,64 @@ sub finalize {
     my ($self) = @_;
     die("finalize called twice!") if ($self->{finalize_called});
     $self->{finalize_called} = 1;
+    my $text = 'Public Cloud finalize: ';
+    diag($text . 'start:');
 
     # Call cleanup() defined in test modules
-    eval { $self->cleanup(); } or record_info('FAILED', "\$self->cleanup() failed -- $@", result => 'fail');
+    eval { $self->cleanup(); } or record_info('FAILED cleanup', $text . "$self->cleanup() failed --\n $@");
 
     my $flags = $self->test_flags();
+    my $upload;
 
-    diag('Public Cloud finalize: $flags->{publiccloud_multi_module}=' . $flags->{publiccloud_multi_module}) if ($flags->{publiccloud_multi_module});
-    diag('Public Cloud finalize: $flags->{fatal}=' . $flags->{fatal}) if ($flags->{fatal});
-    diag('Public Cloud finalize: $self->{result}=' . $self->{result}) if ($self->{result});
-    diag('Public Cloud finalize: $self->{run_args}=' . $self->{run_args}) if ($self->{run_args});
+    diag($text . '$flags->{publiccloud_multi_module}=' . $flags->{publiccloud_multi_module}) if ($flags->{publiccloud_multi_module});
+    diag($text . '$flags->{fatal}=' . $flags->{fatal}) if ($flags->{fatal});
+    diag($text . '$self->{result}=' . $self->{result}) if ($self->{result});
 
     if ($self->{run_args}) {
-        if ($self->{run_args}->{my_instance}) {
-            my $dumpable_instance = Storable::dclone($self->{run_args}->{my_instance});
-            $dumpable_instance->{provider}->{provider_client}->{credentials_file_content} = '******';
-            diag('Public Cloud finalize: $self->{run_args}->{my_instance}=' . Dumper($dumpable_instance));
-        }
+        diag($text . '$self->{run_args}= ' . $self->{run_args});
```

Comment:
DANGER : this should not be merged with this line because here you will output structure which contains password to access Public Cloud API in plain text !

Reply:
correction: no exposure happens because you not using Dumper , but this line does not make much sense in that way `$self->{run_args}= OpenQA::Test::RunArgs=HASH(0x55841c3195e8)`

Reply:
Anyway, it comes to my mind that security should be almost covered already, being such consideration applicable in O3 where no P.C. tested, while OSD context is vpn 🙂

Reply:
it does not matter , I assume you don't want your personal password being available in plain text over some http url even if it is under vpn ,right ?

Reply:
done